pkg/asset/installconfig/azure/client.go

```diff
@@ -328,3 +329,28 @@ func (c *Client) AreMarketplaceImageTermsAccepted(ctx context.Context, publisher
 
 	return terms.AgreementProperties.Accepted != nil && *terms.AgreementProperties.Accepted, nil
 }
+
+// GetAvailabilityZones retrieves a list of availability zones for the given region, and instance type.
+func (c *Client) GetAvailabilityZones(ctx context.Context, instanceType string, region string) ([]string, error) {
+	client := azenc.NewResourceSkusClientWithBaseURI(c.ssn.Environment.ResourceManagerEndpoint, c.ssn.Credentials.SubscriptionID)
+	var zones []string
+	// Only supported filter atm is `location`
+	filter := fmt.Sprintf("location eq '%s'", region)
+	for res, err := client.List(ctx, filter); res.NotDone(); err = res.NextWithContext(ctx) {
+		if err != nil {
+			return zones, err
+		}
+
+		for _, resSku := range res.Values() {
+			if strings.EqualFold(to.String(resSku.Name), instanceType) {
+				for _, locationInfo := range *resSku.LocationInfo {
+					if strings.EqualFold(to.String(locationInfo.Location), region) {
+						return *locationInfo.Zones, nil
+					}
+				}
+			}
+		}
+	}
+
+	return zones, nil
```

Comment:
This follows the original code, so I think it is ok, but it strikes me as odd that this is not an error.

Reply:
For Azure Stack and Single VM regions that do not support availability zones, the Zones field seems to be just an empty slice. For example, `northcentralus` returns:
```
    "locationInfo": [
      {
        "location": "northcentralus",
        "zoneDetails": [],
        "zones": []
      }
    ],
```
So I agree we should probably return some kind of error at this point.